packages/block-editor/src/components/use-on-block-drop/index.js

```diff
@@ -222,12 +243,54 @@ export default function useOnBlockDrop(
 	const registry = useRegistry();
 
 	const insertOrReplaceBlocks = useCallback(
-		( blocks, updateSelection = true, initialPosition = 0 ) => {
+		(
+			blocks,
+			updateSelection = true,
+			initialPosition = 0,
+			clientIdsToReplace = []
+		) => {
+			const clientIds = getBlockOrder( targetRootClientId );
+			const clientId = clientIds[ targetBlockIndex ];
 			if ( operation === 'replace' ) {
-				const clientIds = getBlockOrder( targetRootClientId );
-				const clientId = clientIds[ targetBlockIndex ];
-
 				replaceBlocks( clientId, blocks, undefined, initialPosition );
+			} else if ( operation === 'group' ) {
+				const targetBlock = getBlock( clientId );
+				if ( nearestSide === 'left' ) {
+					blocks.push( targetBlock );
+				} else {
+					blocks.unshift( targetBlock );
+				}
+
+				const groupInnerBlocks = blocks.map( ( block ) => {
+					return createBlock(
+						block.name,
+						block.attributes,
+						block.innerBlocks
+					);
+				} );
+
+				const areAllImages = blocks.every( ( block ) => {
+					return block.name === 'core/image';
+				} );
+
+				const wrappedBlocks = createBlock(
+					areAllImages ? 'core/gallery' : 'core/group',
```

Comment:
Since we're in the block-editor package, do we need to check that the `core/gallery` and `core/group` blocks are available? Just thinking of other 3rd party editors that might wind up using the `block-editor` package.

For the Group block, there's the `isGroupable` selectors, which might be what we're after: https://github.com/WordPress/gutenberg/blob/a8b4cc216c68fa1c42d25e0911d3cdbe8e1e619e/docs/reference-guides/data/data-core-block-editor.md#L1080-L1082

Then, in place of the hard-coded string for `core/group`, we could use `getGroupingBlockName`, similar to how the block actions handles it? https://github.com/WordPress/gutenberg/blob/1710facd57ad89595c6b0dedc5f30cd0f4e694d3/packages/block-editor/src/components/block-actions/index.js#L111

For the Gallery block, I suppose we could do something like `!! getBlockType( 'core/gallery' )` to check that it's available?

Reply:
Oh yes definitely, that's a great point!